### ISSUE TITLE
Global CLI Required

### ISSUE THREAD

**Issue by Lindsor** [ionic-team/ionic-cli#4422]
Currently Ionic requires a bunch of things to be installed globally.
This pattern seems to be very error prone and counter productive.

I run a consultancy company that deals with multiple applications at once running different versions of Ionic (3, 4, 5). 
Depending on the version of the CLI you need different dependencies (cordova-res, native-run, cordova, ...).

Ionic should install these dependencies locally so developers can use `npx <command>` without requiring ANY globally installed packages.

This already currently works to a certain degree, i have an Ionic 3 and 4 project that have everything installed locally and work fine.

This should be the accepted pattern going forward though in order to make this framework more future proof and less frustrating for devs needing to support multiple applications running multiple versions.

**Comment by dwieeb** [comment]
I don't think we require a global installation of anything. You should be able to install `@ionic/cli`, `cordova-res`, `native-run`, and even `cordova` locally and then use `npx ionic <command>`.

The global install is nice for beginners and currently the easiest way to start an app, though we do want to implement `npm init ionic` (see https://github.com/ionic-team/ionic-cli/issues/3300).

**Comment by Lindsor** [comment]
Hey @dwieeb sorry your correct none of it is technically required its just not clear from your docs and from CLI messaged.

What I'm proposing is that this becomes the default behavior both in documentation and in the CLI project generator.
When adding `--cordova` then `cordova` and `cordova-res` should be added to package.json
And the `CLI` should be added to the package.json by default as well
